### DISCUSSION TITLE
Delete functionality

### DISCUSSION THREAD

**Discussion by flowtrader2016** [ramnes/notion-sdk-py#57] [Q&A] [answered]
I have a table and found it relatively easy to create and update the table with row entries (pages) but I'm not sure how to delete these rows.  I would like to delete all rows (pages) every day and add new ones to the table (effectively reset the table each day before adding rows).  Do you have any examples showing similar functionality please?

**Comment by aahnik** [comment, marked as answer]
I think that is not possible as of now because the Notion API does not have any delete endpoint.

**Reply by weallwegot** [reply]
good news - they recently [updated the documentation to include this functionality](https://developers.notion.com/reference/patch-page#archive-delete-a-page)

**Reply by oonatmai** [reply]
@weallwegot Still, when I set the 'archived' value to True, the rows didn't delete itself.
Do you have any suggestion please?

**Reply by ramnes** [reply]
It's because the feature wasn't added in notion-sdk-py. It is now! #66

**Reply by ramnes** [reply]
And I just released 0.5.0 so that you can use it easily. :)

**Reply by bergercookie** [reply]
Is this available in the `BlocksEndpoint` as well?
Setting the `archived` flag to `True` does not move the item out of the page and the returned item from the `update` call still has `archived: False`

**Reply by ramnes** [reply]
@bergercookie Notion API isn't consistent here, and has a DELETE endpoint for blocks that you can use (`BlocksEndpoint.delete()`). But it looks like in the meantime they implemented the support of `"archived"` in the PATCH endpoint as well, so I'm adding it now. :)

**Reply by bergercookie** [reply]
> has a DELETE endpoint for blocks that you can use (BlocksEndpoint.delete())

Ah, thanks, I missed the obvious one! :) 

> But it looks like in the meantime they implemented the support of "archived" in the PATCH endpoint as well, so I'm adding it now. :)

Thanks, and thanks for this awesome sdk too!

**Reply by ramnes** [reply]
Thanks for the kind words! 🥰

And released in 0.8.0, by the way.
